src/EventStore.Core/Services/Transport/Grpc/ServerFeatures.cs

```diff
@@ -5,19 +5,22 @@
 using Grpc.AspNetCore.Server;
 using Grpc.Core;
 using Microsoft.AspNetCore.Routing;
-
+using VersionInfo = EventStore.Common.Utils.VersionInfo;
 namespace EventStore.Core.Services.Transport.Grpc {
 	internal partial class ServerFeatures
 		: EventStore.Client.ServerFeatures.ServerFeatures.ServerFeaturesBase {
 		public const int ApiVersion = 1;
-		private readonly EndpointDataSource _endpointDataSource;
+		private readonly Task<SupportedMethods> _supportedMethods;
```

Comment:
don't think there is a need to cache it as a Task, directly as `SupportedMethods` will do? then the `GetSupportedMethods` that the constructor calls can just return `SupportedMethods`

Reply:
retracted. joao pointed out this would result in allocating a task (from result) when the grpc method is called, so there is an advantage to caching the task